bin/crew

```diff
@@ -1325,8 +1325,9 @@ def resolve_dependencies
 
   # leave only not installed packages in dependencies
   @dependencies.select! {|name| @device[:installed_packages].none? {|pkg| pkg[:name] == name}}
-  ['jdk11','jdk15','jdk16','jdk17'].each do |jdk|
+  ['jdk11','jdk15','jdk16','jdk17','jdk18'].each do |jdk|
```

Comment:
Are you sure that is the correct regex expression?  It's self documenting if each jdk version is listed explicitly.

Reply:
I think this would be better?  `@dependencies.delete('jdk8') if @dependencies.grep(/^jdk8$/).any?`

Reply:
What if there was a package named jdk_extensions that really did need jdk8 as a dependency?  See what I mean?

Reply:
> I think this would be better? `@dependencies.delete('jdk8') if @dependencies.grep(/^jdk8$/).any?`

This regex means "delete jdk8 if jdk8 exists in the dependency list", which is not expected...

> Are you sure that is the correct regex expression? It's self documenting if each jdk version is listed explicitly.

`.grep(/^jdk[^8]/)` will match all jdk version **except** jdk8, delete `jdk8` from the dependency list if the array returned by `.grep` is not empty (either `'jdk11','jdk15','jdk16','jdk17','jdk18'` is exist in the dep list)

> What if there was a package named jdk_extensions that really did need jdk8 as a dependency? See what I mean?

Yeah, that is an issue. But we can avoid matching any non-jdk{11,15,16,17,18} packages by modifying the regex :)
```ruby
/^jdk([1-7]|9)\d+$/ # will return true if there is a jdk version other than `8` exists in deplist
```

Reply:
I think a regex is too aggressive.  We have to assume there is the possibility it will exclude jdk8 when it should be included.

Reply:
I think leaving the regex out makes this much more readable.